app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,11 @@
+class RegistrationsController < Devise::RegistrationsController
```

Comment:
The `RegistrationsController` should not handle your logic for updating billing/shipping.
Please move this logic to another controller.